### ISSUE TITLE
[WEB] Refresh MIME Type Error on destination route when using `usePathUrlStrategy` on Flutter stable 3.24.0

### ISSUE THREAD

**Issue by JAICHANGPARK** [flutter/flutter#152995]
### Steps to reproduce

Hello ~ 
When I add flutter_web_plugins and use usePathUrlStrategy(), an error occurs when I refresh the page.
and When I remove usePathUrlStrategy() and refresh the page, it loads correctly.

I am using Flutter version 3.24.
This was a feature that worked well before.

Thanks 

### Expected results

The page should load correctly when the browser is refreshed.

### Actual results

Refused to execute script from 'http://localhost:58756/home/main.dart.js' because its MIME type ('text/html') is not executable, and strict MIME type checking is enabled.

```
The script has an unsupported MIME type ('text/html').

flutter_bootstrap.js:3 Exception while loading service worker: DOMException: Failed to register a ServiceWorker for scope ('http://localhost:58756/home/') with script ('http://localhost:58756/home/flutter_service_worker.js?v=null'): The script has an unsupported MIME type ('text/html').
(anonymous) @ flutter_bootstrap.js:3
Promise.catch
load @ flutter_bootstrap.js:3
(anonymous) @ flutter_bootstrap.js:12
page1:1 Refused to execute script from 'http://localhost:58756/home/main.dart.js' because its MIME type ('text/html') is not executable, and strict MIME type checking is enabled.
```

## pubspec.yaml
<details open><summary>Code sample</summary>

```yaml

environment:
  sdk: ^3.5.0
dependencies:
  flutter:
    sdk: flutter
  cupertino_icons: ^1.0.8
  go_router: ^14.2.3
  flutter_web_plugins:
    sdk: flutter
```

## main.dart code 
```dart

class Page1 extends StatelessWidget {
  const Page1({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: Text("Page1"),
      ),
    );
  }
}

class Page2 extends StatelessWidget {
  const Page2({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: Text("Page2"),
      ),
    );
  }
}

void main() {
  usePathUrlStrategy(); // 추가
  GoRouter.optionURLReflectsImperativeAPIs = true;
  runApp(MyApp());
}


class MyApp extends StatelessWidget {
  MyApp({super.key});

  final _router = GoRouter(
    routes: [
      GoRoute(
        path: "/",
        builder: (context, state) => const MyHomePage(title: "Home"),
      ),
      GoRoute(path: "/home/page1", builder: (context, state) => Page1()),
      GoRoute(path: "/home/page2", builder: (context, state) => Page2()),
    ],
  );

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp.router(
      title: 'Flutter Demo',
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
        useMaterial3: true,
      ),
      routerConfig: _router,
      // home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key, required this.title});

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;

  void _incrementCounter() {
    setState(() {
      _counter++;
    });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        backgroundColor: Theme.of(context).colorScheme.inversePrimary,
        title: Text(widget.title),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            const Text(
              'You have pushed the button this many times:',
            ),
            Text(
              '$_counter',
              style: Theme.of(context).textTheme.headlineMedium,
            ),
            TextButton(
                onPressed: () {
                  context.go("/home/page1");
                },
                child: Text("페이지 이동"))
          ],
        ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: _incrementCounter,
        tooltip: 'Increment',
        child: const Icon(Icons.add),
      ), 
    );
  }
}
```

</details>

#### Issue Case 
<details open>
<summary>Screenshots / Video demonstration</summary>

https://github.com/user-attachments/assets/e0cfff24-b700-48cb-8d40-4a479ff02cac

</details>

#### default setup case 

https://github.com/user-attachments/assets/136b60d6-3f77-456f-b9dc-f58cc6e766a4

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.24.0, on macOS 14.5 darwin-arm64, locale
    en-KR)
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 15.4)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2024.1)
[✓] IntelliJ IDEA Ultimate Edition (version 2024.1.4)
[✓] VS Code (version 1.92.0)
[✓] Connected device (3 available)
[✓] Network resources
```

</details>

**Comment by JAICHANGPARK** [comment]
here is index.html file

```xml
<!DOCTYPE html>
<html>
<head>
  <!--
    If you are serving your web app in a path other than the root, change the
    href value below to reflect the base path you are serving from.

    The path provided below has to start and end with a slash "/" in order for
    it to work correctly.

    For more details:
    * https://developer.mozilla.org/en-US/docs/Web/HTML/Element/base

    This is a placeholder for base href that will be replaced by the value of
    the `--base-href` argument provided to `flutter build`.
  -->
  <base href="$FLUTTER_BASE_HREF">

  <meta charset="UTF-8">
  <meta content="IE=Edge" http-equiv="X-UA-Compatible">
  <meta name="description" content="A new Flutter project.">

  <!-- iOS meta tags & icons -->
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  <meta name="apple-mobile-web-app-title" content="untitled2">
  <link rel="apple-touch-icon" href="icons/Icon-192.png">

  <!-- Favicon -->
  <link rel="icon" type="image/png" href="favicon.png"/>

  <title>untitled2</title>
  <link rel="manifest" href="manifest.json">
</head>
<body>
  <script src="flutter_bootstrap.js" async></script>
</body>
</html>

```

**Comment by huycozy** [comment]
Hi @JAICHANGPARK 
I can re-create the error (`The script has an unsupported MIME type ('text/html')`) with the following steps:
1. Run app and open it on Web browser:

```console
flutter run -d web-server
```
2. Navigate to next page
3. Click on browser's `Reload` button to refesh the page and observe the error from browser's Devtools console

However, it seems to be fixed on master channel (`3.24.0-1.0.pre.486`) as I don't see it anymore. Can you switch to master channel and confirm?

```
flutter channel master
flutter upgrade
```

**Comment by andrewmlevy** [comment]
We are experiencing this issue as well. Switching to master has fixed the issue. Any idea when this will be pushed to stable?

**Comment by huycozy** [comment]
@andrewmlevy  You can read more about the build release channels here https://github.com/flutter/flutter/blob/master/docs/releases/Flutter-build-release-channels.md.

**Comment by JAICHANGPARK** [comment]
@huycozy Hi 
I checked and tried to change sdk master channel. It was works fine..! 

> Hi @JAICHANGPARK 
> 
> I can re-create the error (`The script has an unsupported MIME type ('text/html')`) with the following steps:
> 
> 1. Run app and open it on Web browser:
> 
> 
> 
> ```console
> 
> flutter run -d web-server
> 
> ```
> 
> 2. Navigate to next page
> 
> 3. Click on browser's `Reload` button to refesh the page and observe the error from browser's Devtools console
> 
> 
> 
> However, it seems to be fixed on master channel (`3.24.0-1.0.pre.486`) as I don't see it anymore. Can you switch to master channel and confirm?
> 
> 
> 
> ```
> 
> flutter channel master
> 
> flutter upgrade
> 
> ```

**Comment by huycozy** [comment]
Updating the issue title since this can be reproduced without `go_router` package as well (sample code https://github.com/flutter/flutter/issues/153179#issue-2458319287)

**Comment by msonntag13** [comment]
If some of you need a temporary solution until this is fixed – the [fix described here](https://stackoverflow.com/a/75335747) seems to work for the current issue as well.
You just need to add the marked lines in your index.html file if you're still using the old templating system, or revert back to it and ignore the deprecation warning:

```
<script>
    window.addEventListener('load', function(ev) {
      // Download main.dart.js
      _flutter.loader.loadEntrypoint({
        entrypointUrl: "/main.dart.js", // <-- THIS LINE
        serviceWorker: {
          serviceWorkerVersion: serviceWorkerVersion,
          serviceWorkerUrl: "/flutter_service_worker.js?v=", // <-- THIS LINE
        },
        onEntrypointLoaded: function(engineInitializer) {
          engineInitializer.initializeEngine().then(function(appRunner) {
            appRunner.runApp();
          });
        }
      });
    });
</script>
```

If you're also using Firebase Messaging for backgound messages in browser, the [provided code sample](https://firebase.google.com/docs/cloud-messaging/flutter/receive#web) for the old templating system works out of the box as well:
```
<html>
<body>
  <script>
      var serviceWorkerVersion = null;
      var scriptLoaded = false;
      function loadMainDartJs() {
        if (scriptLoaded) {
          return;
        }
        scriptLoaded = true;
        var scriptTag = document.createElement('script');
        scriptTag.src = 'main.dart.js';
        scriptTag.type = 'application/javascript';
        document.body.append(scriptTag);
      }

      if ('serviceWorker' in navigator) {
        // Service workers are supported. Use them.
        window.addEventListener('load', function () {
          // Register Firebase Messaging service worker.
          navigator.serviceWorker.register('firebase-messaging-sw.js', {
            scope: '/firebase-cloud-messaging-push-scope',
          });

          // Wait for registration to finish before dropping the <script> tag.
          // Otherwise, the browser will load the script multiple times,
          // potentially different versions.
          var serviceWorkerUrl =
            'flutter_service_worker.js?v=' + serviceWorkerVersion;

          navigator.serviceWorker.register(serviceWorkerUrl).then((reg) => {
            function waitForActivation(serviceWorker) {
              serviceWorker.addEventListener('statechange', () => {
                if (serviceWorker.state == 'activated') {
                  console.log('Installed new service worker.');
                  loadMainDartJs();
                }
              });
            }
            if (!reg.active && (reg.installing || reg.waiting)) {
              // No active web worker and we have installed or are installing
              // one for the first time. Simply wait for it to activate.
              waitForActivation(reg.installing ?? reg.waiting);
            } else if (!reg.active.scriptURL.endsWith(serviceWorkerVersion)) {
              // When the app updates the serviceWorkerVersion changes, so we
              // need to ask the service worker to update.
              console.log('New service worker available.');
              reg.update();
              waitForActivation(reg.installing);
            } else {
              // Existing service worker is still good.
              console.log('Loading app from service worker.');
              loadMainDartJs();
            }
          });

          // If service worker doesn't succeed in a reasonable amount of time,
          // fallback to plaint <script> tag.
          setTimeout(() => {
            if (!scriptLoaded) {
              console.warn(
                'Failed to load app from service worker. Falling back to plain <script> tag.'
              );
              loadMainDartJs();
            }
          }, 4000);
        });
      } else {
        // Service workers not supported. Just drop the <script> tag.
        loadMainDartJs();
      }
  </script>
</body>
```

**Comment by jsaulsberry-cvet** [comment]
Seeing this issue as well

**Comment by katjai1** [comment]
loadEntrypoint do not need anymore, "load" is working well, again.

**Comment by huycozy** [comment]
@JAICHANGPARK (and also anyone facing this issue) I checked this on the new stable hotfix 3.24.1 and didn't see the issue anymore, can you please confirm this as well?

**Comment by msonntag13** [comment]
I can confirm as well, works fine with 3.24.1

**Comment by katjai1** [comment]
Yes, actually fixed in 3.24.1

**Comment by huycozy** [comment]
Thank you for confirming this. I'm going ahead and closing this as fixed.

**Comment by nfairfield** [comment]
Unfortunately, it doesn't seem to be fixed... or perhaps it has revealed another issue.

* I upgraded to flutter 3.24.2, and switched from the old-fashioned 'loadEntrypoint' workaround to the new 'flutter_bootstrap.js' method
* Safari (macbook and ios) started just giving blank white pages
 
It *looks* like there is some sort of problem with canvaskit.js, import.meta statements only being valid in modules, etc.
When I rolled back to the old 'loadEntrypoint' method, Safari started working again.
